src/python/pants/reporting/zipkin_reporter.py

```diff
@@ -191,3 +204,129 @@ def bulk_record_workunits(self, engine_workunits):
 
 def from_secs_and_nanos_to_float(secs, nanos):
   return secs + ( nanos / NANOSECONDS_PER_SECOND )
+
+
+def emit_spans(self):
+
+  if not self.zipkin_attrs.is_sampled:
+    self._get_tracer().clear()
+    return
+
+  end_timestamp = time.time()
+
+  annotations = {}
+  if self.add_logging_annotation:
+    annotations[LOGGING_END_KEY] = time.time()
+
+  last_span = Span(
+    trace_id=self.zipkin_attrs.trace_id,
+    name=self.span_name,
+    parent_id=self.zipkin_attrs.parent_span_id,
+    span_id=self.zipkin_attrs.span_id,
+    kind=Kind.CLIENT if self.client_context else Kind.SERVER,
+    timestamp=self.start_timestamp,
+    duration=end_timestamp - self.start_timestamp,
+    local_endpoint=self.endpoint,
+    remote_endpoint=self.remote_endpoint,
+    shared=not self.report_root_timestamp,
+    annotations=annotations,
+    tags=self.tags,
+  )
+  self._get_tracer().add_span(last_span)
+
+  safe_mkdir(self.zipkin_spans_dir)
+
+  spans_sender = ZipkinSpanSender(self.transport_handler,
+    self.max_span_batch_size,
+    self.encoder,
+    self.zipkin_spans_dir,
+    self.encoding)
+
+  with spans_sender:
+    if os.path.exists(self.zipkin_spans_dir):
+      for span in self._get_tracer().get_spans():
+        span.local_endpoint = copy_endpoint_with_new_service_name(
+          self.endpoint,
+          span.local_endpoint.service_name,
+        )
+
+        spans_sender.add_span(span)
+
+  self._get_tracer().clear()
+
+
+class ZipkinSpanSender(object):
+
+  MAX_BATCH_SIZE = 100
+
+  def __init__(self, transport_handler, max_span_batch_size, encoder, zipkin_spans_dir, encoding):
```

Comment:
I'd add to a docstring here that the encoder/encoding _must_ be JSON (because the class assumes that it is)